tools/dtest/config/config.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/m3db/m3em/node"
+	"github.com/m3db/m3x/config"
+	xlog "github.com/m3db/m3x/log"
+	"google.golang.org/grpc"
+)
+
+var (
+	defaultConnectionTimeout = 2 * time.Minute
+)
+
+// Configuration is a collection of knobs to control test behavior
+type Configuration struct {
+	DTest DTestConfig              `yaml:"dtest"`
+	M3EM  M3EMConfig               `yaml:"m3em"`
+	KV    etcdclient.Configuration `yaml:"kv"`
+}
+
+// TODO(prateek): merge DTest/M3EM
+// TODO(prateek): make config structs for options in m3em
+
+// M3EMConfig is a list of m3em environment settings
+type M3EMConfig struct {
+	HeartbeatPort int                 `yaml:"heartbeatPort" validate:"nonzero"`
+	AgentPort     int                 `yaml:"agentPort" validate:"nonzero"`
+	M3DBPort      int                 `yaml:"m3dbPort" validate:"nonzero"`
+	M3DBServiceID string              `yaml:"m3dbServiceID" validate:"nonzero"`
+	Instances     []PlacementInstance `yaml:"instances" validate:"min=1"`
+	Node          node.Configuration  `yaml:"node"`
+}
+
+// DTestConfig is a collection of DTest configs
+type DTestConfig struct {
+	DebugPort            int `yaml:"debugPort" validate:"nonzero"`
+	NumShards            int `yaml:"numShards" validate:"nonzero"`
+	BootstrapTimeoutMins int `yaml:"bootstrapTimeoutMins" validate:"nonzero"`
+}
+
+// PlacementInstance is a config for a services.PlacementInstance
+type PlacementInstance struct {
+	ID       string `yaml:"id" validate:"nonzero"`
+	Rack     string `yaml:"rack" validate:"nonzero"`
+	Zone     string `yaml:"zone" validate:"nonzero"`
+	Weight   uint32 `yaml:"weight" validate:"nonzero"`
+	Hostname string `yaml:"hostname" validate:"nonzero"`
+}
+
+// New constructs a Configuration object from the path specified
+func New(m3emConfigPath string) (*Configuration, error) {
+	var conf Configuration
+	if err := xconfig.LoadFile(&conf, m3emConfigPath); err != nil {
+		return nil, err
+	}
+
+	return &conf, nil
+}
+
+// M3DBNodes returns a slice of m3dbnode.Nodes per the config provided
+func (mc *M3EMConfig) M3DBNodes(opts node.Options, numNodes int) ([]m3dbnode.Node, error) {
+	// use all nodes if numNodes is zero
+	if numNodes <= 0 {
+		numNodes = len(mc.Instances)
+	}
+
+	var (
+		logger  = opts.InstrumentOptions().Logger()
+		nodes   = make([]m3dbnode.Node, 0, len(mc.Instances))
+		nodeNum = 0
+	)
+
+	for _, inst := range mc.Instances {
+		if nodeNum >= numNodes {
+			break
+		}
+
+		pi := inst.newServicesPlacementInstance(mc.M3DBPort)
+		newOpts := opts.
+			SetOperatorClientFn(inst.operatorClientFn(mc.AgentPort)).
+			SetInstrumentOptions(opts.InstrumentOptions().SetLogger(
+				logger.WithFields(xlog.NewLogField("host", inst.Hostname))))
+
+		svcNode, err := node.New(pi, newOpts)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create service node for %+v, error: %v", inst, err)
+		}
+
+		m3dbNodeOpts := m3dbnode.NewOptions(newOpts.InstrumentOptions()).SetNodeOptions(newOpts)
+		m3dbNode, err := m3dbnode.New(svcNode, m3dbNodeOpts)
```

Comment:
Naming of this vars too, I feel like we can get rid of a lot of "M3DB" prefixed vars and config.